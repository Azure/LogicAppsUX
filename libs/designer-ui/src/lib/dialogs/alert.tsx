import type { IDialogContentProps, IModalProps } from '@fluentui/react';
import { Dialog, DialogFooter, PrimaryButton } from '@fluentui/react';
import { useIntl } from 'react-intl';

export interface AlertProps {
  hidden: boolean;
  message: string;
  title: string;
  onDismiss(): void;
}

const modalProps: IModalProps = {
  isBlocking: true,
  firstFocusableSelector: 'dialog-ok-button',
};

export const Alert: React.FC<AlertProps> = ({ hidden, message, title, onDismiss }) => {
  const intl = useIntl();
  if (hidden) {
    return null;
  }

  const dialogContentProps: IDialogContentProps = {
    title,
  };

  const okMessage = intl.formatMessage({
    defaultMessage: 'OK',
    description: 'OK message appearing on a alert message modal.',
  });
  return (
    <Dialog dialogContentProps={dialogContentProps} hidden={hidden} modalProps={modalProps} onDismiss={onDismiss}>
      {message}
      <DialogFooter>
        <PrimaryButton className="dialog-ok-button" onClick={onDismiss}>
          {okMessage}
        </PrimaryButton>
      </DialogFooter>
    </Dialog>
  );
};
