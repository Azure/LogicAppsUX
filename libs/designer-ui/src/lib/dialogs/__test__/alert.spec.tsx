import * as React from 'react';
import * as ReactShallowRenderer from 'react-test-renderer/shallow';

import { Alert, AlertProps } from '../alert';

describe('ui/dialogs/_alert', () => {
  let minimal: AlertProps;
  let renderer: ReactShallowRenderer.ShallowRenderer;
  beforeEach(() => {
    minimal = {
      hidden: false,
      message: 'Message',
      title: 'Title',
      onDismiss: jest.fn(),
    };
    renderer = ReactShallowRenderer.createRenderer();
  });

  it('should render', () => {
    const alert = renderer.render(<Alert {...minimal} />);
    expect(alert).toMatchSnapshot();
  });
});
