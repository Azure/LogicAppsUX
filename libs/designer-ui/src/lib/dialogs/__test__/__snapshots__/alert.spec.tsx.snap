// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/dialogs/_alert should render 1`] = `
<Dialog
  dialogContentProps={
    Object {
      "title": "Title",
    }
  }
  hidden={false}
  modalProps={
    Object {
      "firstFocusableSelector": "dialog-ok-button",
      "isBlocking": true,
    }
  }
  onDismiss={[MockFunction]}
>
  Message
  <StyledDialogFooterBase>
    <CustomizedPrimaryButton
      className="dialog-ok-button"
      onClick={[MockFunction]}
    >
      <Memo(MemoizedFormattedMessage)
        defaultMessage={
          Array [
            Object {
              "type": 0,
              "value": "OK",
            },
          ]
        }
        id="uN4zFU"
      />
    </CustomizedPrimaryButton>
  </StyledDialogFooterBase>
</Dialog>
`;
